dalle2_pytorch/dalle2_pytorch.py

```diff
@@ -679,15 +710,31 @@ def sample(self, text, num_samples_per_batch = 2):
         top_image_embeds = image_embeds.gather(1, top_sim_indices)
         return rearrange(top_image_embeds, 'b 1 d -> b d')
 
-    def forward(self, text, image, *args, **kwargs):
-        b, device, img_size, = image.shape[0], image.device, self.image_size
-        check_shape(image, 'b c h w', h = img_size, w = img_size, c = self.channels)
+    def forward(self, text, image, text_embed=None, image_embed=None, *args, **kwargs):
+        # check if we are doing embedding-only training
+        if text is not None:
+            b, device, img_size, = image.shape[0], image.device, self.image_size
+            check_shape(image, 'b c h w', h=img_size,
+                        w=img_size, c=self.channels)
+            text_cond = self.get_text_cond(text)
+        else:
+            b, device = image_embed.shape[0], image_embed.device
```

Comment:
The DiffusionPriorNetwork expects a dictionary. So I format the predefined text embedding in a dictionary with text_encodings:None to make it possible to detect the embedding-only training once inside the DiffusionPriorNetwork.

Reply:
yea, so here i'd actually just do `def forward(self, text = None, image = None, text_embed = None, image_embed = None)`

and later on do asserts to make sure one or the other is passed in `assert exists(text) ^ exists(text_embed)` and `assert exists(image) ^ exists(image_embed)` and handle them both in separate conditionals, prepping the kwargs for the `p_losses` function below